src/Server/Helper.php

```diff
@@ -79,17 +81,17 @@ public function parseHttpRequest(?callable $readRawBodyFn = null)
             }
 
             if (stripos($contentType, 'application/graphql') !== false) {
-                $rawBody    = $readRawBodyFn
-                    ? $readRawBodyFn()
-                    : $this->readRawBody();
+                $rawBody    = $readRawBodyFn === null
+                    ? $this->readRawBody()
+                    : $readRawBodyFn();
                 $bodyParams = ['query' => $rawBody ?? ''];
             } elseif (stripos($contentType, 'application/json') !== false) {
-                $rawBody    = $readRawBodyFn ?
-                    $readRawBodyFn()
-                    : $this->readRawBody();
+                $rawBody    = $readRawBodyFn === null
+                    ? $this->readRawBody()
+                    : $readRawBodyFn();
                 $bodyParams = json_decode($rawBody ?? '', true);
 
-                if (json_last_error()) {
+                if (json_last_error() !== JSON_ERROR_NONE) {
```

Comment:
Off-topic: We could consider using https://github.com/thecodingmachine/safe.

Reply:
Not sure if worth though. There's some lack of static analysis tools support / it's often behind with updates. 

Eg. there are some dynamic extensions in phpstan/psalm that are not working for Safe and must be ported there. Also this example https://github.com/hectorj/safe-php-psalm-plugin/pull/5. 

On my private projects I use it but for libraries I'm trying to consider these cons.